cli/dts/lib.deno.ns.d.ts

```diff
@@ -3029,4 +3032,136 @@ declare namespace Deno {
     recordType: RecordType,
     options?: ResolveDnsOptions,
   ): Promise<string[] | MXRecord[] | SRVRecord[] | string[][]>;
+
+  export interface CommandOptions {
+    /** Arguments to pass to the process. */
+    args?: string[];
+    /**
+     * The working directory of the process.
+     * If not specified, the cwd of the parent process is used.
+     */
+    cwd?: string | URL;
+    /**
+     * Clear environmental variables from parent process.
+     * Doesn't guarantee that only `opt.env` variables are present,
+     * as the OS may set environmental variables for processes.
+     */
+    clearEnv?: boolean;
+    /** Environmental variables to pass to the subprocess. */
+    env?: Record<string, string>;
+    /**
+     * Sets the child process’s user ID. This translates to a setuid call
+     * in the child process. Failure in the setuid call will cause the spawn to fail.
+     */
+    uid?: number;
+    /** Similar to `uid`, but sets the group ID of the child process. */
+    gid?: number;
+
+    stdin?: "piped" | "inherit" | "null";
+    stdout?: "piped" | "inherit" | "null";
+    stderr?: "piped" | "inherit" | "null";
+  }
+
+  /**
+   * Spawns a child process.
+   *
+   * `CommandOptions.stdin` defaults to `"null"`.
+   * `CommandOptions.stdout` defaults to `"inherit"`.
+   * `CommandOptions.stderr` defaults to `"inherit"`.
+   *
+   * ```ts
+   * const child = Deno.spawnChild(Deno.execPath(), {
+   *   args: [
+   *     "eval",
+   *     "console.log('Hello World')",
+   *   ],
+   * });
+   *
+   * const status = await child.wait();
+   * ```
+   */
+  export function spawnChild<T extends CommandOptions = CommandOptions>(
+    command: string | URL,
+    options?: T,
+  ): Child<T>;
+
+  export class Child<T extends CommandOptions> {
+    readonly stdin: T["stdin"] extends "piped" ? WritableStream<Uint8Array>
+      : null;
+    readonly stdout: T["stdout"] extends "piped" ? ReadableStream<Uint8Array>
+      : null;
+    readonly stderr: T["stderr"] extends "piped" ? ReadableStream<Uint8Array>
+      : null;
+
+    readonly pid: number;
+    /** Get the current status. */
+    readonly status: CommandStatus | null;
+
+    /** Waits for the child to exit completely, returning the status that it exited with. */
+    wait(): Promise<CommandStatus>;
```

Comment:
I still don't like this as I had elaborated in earlier issue threads and comments on this PR. I think this should be `status: Promise<CommandStatus>`. My suggestion has web API precedent (`closed` field on `WebSocketStream`), whereas the currently implemented API does not.

Reply:
someone deep into this comment section I have explained: this is not possible. 
> The stdin handle to the child process, if any, will be closed before waiting. This helps avoid deadlock: it ensures that the child does not block waiting for input from the parent, while the parent waits for the child to exit.

from the tokio docs for [Child.wait()](https://docs.rs/tokio/latest/tokio/process/struct.Child.html#method.wait), which is different behaviour to `closed` field. a method makes this more intuitive that it will have a side-effect

Reply:
I think we shouldn't have this side effect - see comments below. This is also not an explicit operation in Node, or in most other runtimes.